### PR TITLE
feat: control-objective-selection

### LINKED ISSUE #639: SAP: Test Case Procedures
**Extended Description**
- As an ASAP developer, in order to assist FedRAMP Reviewers in quickly identifying problems with objectives, I want to verify that included and excluded objectives within the SAP are identified as available within the associated catalog.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- [x] exclude-objective @objective-id values do not match any include-objective @objective-id values.
- [x] exclude-objective @objective-id values are available in the associated catalog
- [x] include-objective @objective-id values are available in the associated catalog

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%
 - [x] Automated code quality checks passed
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed – needs to be created/automated
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
close #639 
asserts, diagnostic, xspec for:
* include-all or include-objective
* duplicate checking for exclude- or include- objectives @objective-id values
asserts and diagnostics for:
* Compare to profile objectives with SAP objective values.

Minor formatting

### CODE REVIEW

File: src/validations/rules/sap.sch [hunk 1]
```diff
@@ -319,6 +319,96 @@
...
+                diagnostics="duplicate-exclude-objective-values-diagnostic"
+                id="duplicate-exclude-objective-values"
+                role="error"
+                test="count($exclude-control-objective-ids) eq count(distinct-values($exclude-control-objective-ids))">The
+                exclude-objective/@objective-id values must not be duplicated.</sch:assert>
+        </sch:rule>
+        <!-- This series of variables gets to the profile file to match the include and exclude asserts below -->
+        <sch:let
+            name="profile-href"
+            value="resolve-uri($ssp-doc/oscal:system-security-plan/oscal:import-profile/@href, base-uri())" />
```
Comment (GaryGapinski) [inline src/validations/rules/sap.sch:369]: An SSP `import-profile` may not be importing a profile.

File: src/validations/rules/sap.sch [hunk 2]
```diff
@@ -319,6 +319,96 @@
...
+                role="error"
+                test="count($include-control-objective-ids) eq count(distinct-values($include-control-objective-ids))">The
+                include-objective/@objective-id values must not be duplicated.</sch:assert>
+
+            <sch:assert
+                diagnostics="duplicate-exclude-objective-values-diagnostic"
+                id="duplicate-exclude-objective-values"
+                role="error"
+                test="count($exclude-control-objective-ids) eq count(distinct-values($exclude-control-objective-ids))">The
+                exclude-objective/@objective-id values must not be duplicated.</sch:assert>
```
Comment (GaryGapinski) [inline src/validations/rules/sap.sch:364]: These will unfortunately not identify the actual duplicates. To precisely identify duplicates would require assertions on `include-objective` and `exclude-objective`.

**Review by GaryGapinski — APPROVED** [review]

File: src/validations/rules/sap.sch (hunk 1 above)
Comment (markXLIX) [reply src/validations/rules/sap.sch:369]: Made changes to import-profile handling to check for oscal:profile element in referenced document.  At least this way the excessive errors due to missing objectives will be easier to track down for the user.

File: src/validations/rules/sap.sch (hunk 2 above)
Comment (markXLIX) [reply src/validations/rules/sap.sch]: Yes, this test doesn't isolate the @objective-id values.  It just states that there are duplicates within the set of those values for each element name.  

I've changed the assertions to address each include-objective and exclude-objective element separately for these tests.

**Pull request merged**